client/lib/route-modal/use-route-modal.tsx

```diff
@@ -0,0 +1,47 @@
+import { addQueryArgs, removeQueryArgs } from '@wordpress/url';
+import page from 'page';
+import { useSelector } from 'react-redux';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import getPreviousRoute from 'calypso/state/selectors/get-previous-route.js';
+
+interface ReturnValues {
+	/** Whether the query key exists or not */
+	isModalOpen: boolean;
+	/** The value of query key */
+	value: any;
+	/** Set the query key to value */
+	openModal: ( value?: any ) => void;
+	/** Clears the query key */
+	closeModal: () => void;
+}
+
+/**
+ * React hook providing utils to control opening and closing modal via query string.
+ */
+const useRouteModal = ( queryKey: string, defaultValue: any = '' ): ReturnValues => {
```

Comment:
Can we change `any` to `unknown` for all `*value` args here?

Reply:
Okay